torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -0,0 +1,848 @@
+            [
+                first_dim,
+                *args[2:],
+            ],
+            tx.output.graph,
+            checkpoint,
+        )
+
+        body_nn_modules = tx.copy_graphstate().output.nn_modules
+
+        body_name = add_subgraph(
+            tx,
+            self.source,
+            "map_body",
+            torch.fx.GraphModule(body_nn_modules.nn_modules, body_graph),
+        )
+
+        body_node = make_attr(tx, body_name)
+        p_args = (
+            body_node,
+            *(arg.as_proxy() for arg in args[1:]),
+            *(arg for arg in body_lifted_freevars.keys()),
+        )
+        r = body_r.as_proxy().node.meta["example_value"]
+        example_value = r.new_empty(
+            [get_fake_value(args[1].as_proxy().node, tx).shape[0], *r.shape]
+        )
+
+        _, p_kwargs = proxy_args_kwargs([], kwargs)
+
+        # Store the invocation as a call
+        return wrap_fx_proxy(
+            tx=tx,
+            proxy=tx.output.create_proxy(
+                "call_function",
+                self.value,
+                args=tuple(p_args),
+                kwargs=p_kwargs,
+            ),
+            example_value=example_value,
+        )
+
+
+class ExecutorchCallDelegateHigherOrderVariable(TorchHigherOrderOperatorVariable):
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        from . import ConstantVariable
+        from .builder import wrap_fx_proxy
+
+        self.check_kwargs(kwargs, ConstantVariable)
+
+        # This is operator for delegation within Executorch which calls a
+        # specific function in the given lowered module with the given
+        # operators. The actual operator is defined in the Executorch codebase.
+        # This is a bad hierarchical violation since
+        # executorch_call_delegate sits at a higher level than dynamo, but
+        # there's no real solution to this issue yet.
+        lowered_module = tx.output.get_submodule(args[0].module_key)
+
+        lowered_node = make_attr(tx, args[0].module_key)
+
+        p_args = tuple(arg.as_proxy() for arg in args[1:])
+        real_sub_args = pytree.tree_map_only(
+            torch.fx.Proxy, lambda a: get_real_value(a.node, tx.output), p_args
+        )
+        example_res = lowered_module.original_module(*real_sub_args)
+        example_value = deepcopy_to_fake_tensor(example_res, tx.fake_mode)
+
+        p_args = (lowered_node,) + p_args
+
+        _, p_kwargs = proxy_args_kwargs([], kwargs)
+
+        # Store the invocation as a call
+        return wrap_fx_proxy(
+            tx=tx,
+            proxy=tx.output.create_proxy(
+                "call_function",
+                self.value,
+                args=tuple(p_args),
+                kwargs=p_kwargs,
+            ),
+            example_value=example_value,
+        )
+
+
+class HigherOrderFuncTorchGradVariable(TorchHigherOrderOperatorVariable):
```

Comment:
nit: FuncTorchGradHigherOrderVariable

Reply:
Ah, missed this one. Thanks for catching it!